src/cow-react/modules/wallet/web3-react/connectors/Ledger/provider.tsx

```diff
@@ -0,0 +1,154 @@
+import { TypedDataUtils, SignTypedDataVersion } from '@metamask/eth-sig-util'
+import { TypedDataDomain, TypedDataField } from '@ethersproject/abstract-signer'
+import type Transport from '@ledgerhq/hw-transport'
+import ProviderEngine from 'web3-provider-engine'
+import { LedgerOptions } from './connector'
+
+global.process = { browser: true } as any
+const RpcSubprovider = require('web3-provider-engine/subproviders/rpc')
+
+interface LedgerModules {
+  transportFactory: {
+    create(): Promise<Transport>
+  }
+  createLedgerSubprovider: (factory: () => Promise<Transport>, options: unknown) => any
+  LedgerEth: new (transport: Transport) => any
+}
+
+export interface EcdsaSignature {
+  r: string
+  s: string
+  v: number
+}
+
+// TODO: use current chain Id
+const currentChainId = 1
```

Comment:
Please, don't forget about this

Reply:
Changed it to pass it from the connector constructor or use default.